cpp/src/arrow/filesystem/hdfs.cc

```diff
@@ -268,7 +268,7 @@ void HdfsOptions::ConfigureEndPoint(const std::string& host, int port) {
   connection_config.port = port;
 }
 
-void HdfsOptions::ConfigureHdfsDriver(bool use_hdfs3) {
+void HdfsOptions::ConfigureHdfs3Driver(bool use_hdfs3) {
```

Comment:
I don't think there's any point in changing this method name.

Reply:
Ok. I will put it back as not related to this PR. 

Can we rename this in anther review ? The name of the method is not consistent wit hwhat it is doing. It is called `ConfigureHdfsDriver` whereas what it does is activating the hdfs3 driver. 
libhdfs and hdfs3 are both drivers for hdfs. Libhdfs is activated by default and this method activates hdfs3 driver.

Reply:
I disagree, the naming is correct. This method configures either hdfs driver depending on its argument.

Reply:
As you wish. Let's keep this as is but I also disagree and prefer `ConfigureHdfs3Driver`
```
void HdfsOptions::ConfigureHdfsDriver(bool use_hdfs3) {
  if (use_hdfs3) {
    connection_config.driver = ::arrow::io::HdfsDriver::LIBHDFS3;
  } else {
    connection_config.driver = ::arrow::io::HdfsDriver::LIBHDFS;
  }
}
```
ConfigureHdfsDriver(true) -> means use driver hdfs3
ConfigureHdfsDriver(false) -> means use driver libhdfs (we still use a hdfs driver here)

I've put back the initial naming.

Reply:
Nevermind for the comment. This is a actually a detail.